File: src/main/java/com/qtech/bubbles/common/random/EntityRandomizer.java
package com.qtech.bubbles.common.random;

import com.qtech.bubbles.common.EntityProperties;
import com.qtech.bubbles.common.gametype.AbstractGameType;

import java.awt.geom.Rectangle2D;

/**
 * Abstract Entity Randomizer.
 */
public abstract class EntityRandomizer {
    public abstract EntityProperties getRandomProperties(Rectangle2D bounds, AbstractGameType gameType);
}

File: src/main/kotlin/com/qtech/bubbles/common/random/EntityRandomizer.kt
package com.qtech.bubbles.common.random

import com.qtech.bubbles.common.EntityProperties
import com.qtech.bubbles.common.gametype.AbstractGameMode
import java.awt.geom.Rectangle2D

/**
 * Abstract Entity Randomizer.
 */
abstract class EntityRandomizer {
    abstract fun getRandomProperties(bounds: Rectangle2D, gameMode: AbstractGameMode): EntityProperties
}